File: serve.js
// node serve.js

import fs from 'fs'
import http from 'http'

let folder = 'a'
let port = 1234
let types = {
  css: 'text/css',
  html: 'text/html',
  ico: 'image/x-icon',
  js: 'application/javascript',
  json: 'application/json'
}

function prepare (request, response) {
  let file = folder + request.url
  let path = fs.existsSync(file) && fs.statSync(file).isFile() ? file : folder + '/x.html'
  let type = types[path.split('.').pop()]
  if (type) response.writeHead(200, {'content-type': type})
  fs.createReadStream(path).pipe(response)
}

http.createServer((request, response) => prepare(request, response)).listen(port)

console.log('localhost:' + port)

File: serve.py
# python3 serve.py

from http.server import SimpleHTTPRequestHandler as handler
from socketserver import TCPServer as server

folder = 'a'
port = 1234
types = {
    'css': 'text/css',
    'html': 'text/html',
    'ico': 'image/x-icon',
    'js': 'application/javascript',
    'json': 'application/json'
}

class prepare (handler):
    def do_GET (self):
        try:
            ftype = self.path.split('.')[-1]
            if ftype in types:
                self.send_response(200)
                self.send_header('content-type', types[ftype])
                self.end_headers()
                with open(folder + self.path, 'rb') as file:
                    self.wfile.write(file.read())
                return
        except:
            next
        self.path = folder + '/x.html'
        return handler.do_GET(self)
    def log_message (*args):
        pass

server.allow_reuse_address = True
serve = server(('', port), prepare)
print('localhost:' + str(port))
serve.serve_forever()